src/libpriv/rpmostree-kernel.cxx

```diff
@@ -85,11 +86,12 @@ find_kernel_and_initramfs_in_bootdir (int rootfs_dfd, const char *bootdir, char
 
       name = dent->d_name;
 
-      /* Current Fedora 23 kernel.spec installs as just vmlinuz */
-      if (g_str_equal (name, "vmlinuz") || g_str_has_prefix (name, "vmlinuz-"))
+      if (out_ksuffix ? g_str_has_prefix (name, "vmlinuz-") : g_str_equal (name, "vmlinuz"))
         {
           if (ret_kernel)
-            return glnx_throw (error, "Multiple vmlinuz- in %s", bootdir);
+            return glnx_throw (error, "Multiple vmlinuz%s in %s", out_ksuffix ? "-" : "", bootdir);
+          if (out_ksuffix)
+            ret_ksuffix = g_strdup (name + (sizeof ("vmlinuz-") - 1));
```

Comment:
I'd spell this as just `strlen ("vmlinuz-")` which is less obscure; `sizeof ("vmlinuz-") - 1` requires C programmers to remember that `sizeof` includes the `NUL` character and you're subtracting it off.